src/libs/data-utils.js

```diff
@@ -340,3 +356,231 @@ export const EntityUploader = class EntityUploader extends Component {
+                onChange: () => setEditType(type),
+                labelStyle: { paddingLeft: '0.5rem' }
+              })])
+            ]), [
+              { type: 'string' },
+              { type: 'reference', tooltip: 'A link to another entity' },
+              { type: 'number' },
+              { type: 'boolean' }
+            ]))
+          ]),
+          editType === 'reference' && div({ style: { marginTop: '0.5rem' } }, [
+            div({ style: { marginBottom: '0.5rem' } }, 'Referenced entity type:'),
+            h(Select, {
+              value: linkedEntityType,
+              options: entityTypes,
+              onChange: ({ value }) => setLinkedEntityType(value)
+            })
+          ])
+        ]),
+        div({ style: { marginBottom: '0.5rem' } }, [
+          h(LabeledCheckbox, {
+            checked: isList,
+            onChange: willBeList => setNewValue(willBeList ? [newValue] : newValue[0])
+          }, [span({ style: { marginLeft: '0.5rem' } }, ['Attribute is a list'])])
+        ]),
+        isList ?
+          div({ style: { marginTop: '1.5rem' } }, _.map(([i, value]) => div({
+            style: { display: 'flex', alignItems: 'center', marginBottom: '0.5rem' }
+          }, [
+            makeInput({
+              'aria-label': `List value ${i + 1}`,
+              autoFocus: true,
+              placeholder: 'Enter a value',
+              value,
+              onChange: v => setNewValue([...newValue.slice(0, i), v, ...newValue.slice(i + 1)])
+            }),
+            h(Link, {
+              'aria-label': `Remove list value ${i + 1}`,
+              disabled: newValue.length === 1,
+              onClick: () => setNewValue([...newValue.slice(0, i), ...newValue.slice(i + 1)]),
+              style: { marginLeft: '0.5rem' }
+            }, [
+              icon('times', { size: 20 })
+            ])
+          ]), Utils.toIndexPairs(newValue))) :
+          div({ style: { marginTop: '1.5rem' } }, [
+            makeInput({
+              'aria-label': 'New value',
+              autoFocus: true,
+              placeholder: 'Enter a value',
+              value: newValue,
+              onChange: setNewValue
+            })
+          ]),
+        isList && h(Link, {
+          style: { display: 'block', marginTop: '1rem' },
+          onClick: () => setNewValue(Utils.append(''))
+        }, [icon('plus', { style: { marginRight: '0.5rem' } }), 'Add item']),
+        div({ style: { marginTop: '2rem', display: 'flex', alignItems: 'baseline' } }, [
+          h(ButtonOutline, { onClick: () => setConsideringDelete(true) }, ['Delete']),
+          div({ style: { flexGrow: 1 } }),
+          h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: onDismiss }, ['Cancel']),
+          h(ButtonPrimary, {
+            onClick: doEdit,
+            disabled: _.isEqual(attributeValue, newValue) || newValue === undefined,
+            tooltip: _.isEqual(attributeValue, newValue) && 'No changes to save'
```

Comment:
This check doesn't catch changing the data type or reference flag without changing the value.